Added some basic inputs

JulianStremel opened the pull request:
i am by no stretch of the imagination a good programmer but i hope the additions can help some people.

Added Netflix/Amazon/YouTube support via --key (netflix/amazon/youtube)
Added more Buttons (ok/play/pause/fast_forward/etc.)
Added input selection via --key (source_1/source_2..../source_7)

JulianStremel committed:
Add files via upload

Added Netflix/Amazon/YouTube support via --key (netflix/amazon/youtube)
Added more Buttons (ok/play/pause/fast_forward/etc.)
Added input selection via --key (source_1/source_2....)

JulianStremel committed:
Added some additional features

Added Netflix/Amazon/YouTube support via --key (netflix/amazon/youtube)
Added more Buttons (ok/play/pause/fast_forward/etc.)
Added input selection via --key (source_1/source_2....)

JulianStremel committed:
Add files via upload

newAM committed:
Format files with black.

newAM commented on hisensetv/__init__.py:
```diff
@@ -226,6 +239,60 @@ def send_key(self, keyname: str):
         """
         self._call_service(service="remote_service", action="sendkey", payload=keyname)
 
+    def launch_app(self, app: str):
```
Can you add a docstring here, or prefix with `_` to indicate that this is an internal function?

newAM commented on hisensetv/__init__.py:
```diff
@@ -226,6 +239,60 @@ def send_key(self, keyname: str):
+            }
+
+        else:
+            pass
```
```suggestion
            raise ValueError(f"{app} is not a known app.")
```

newAM commented on hisensetv/__init__.py:
```diff
@@ -226,6 +239,60 @@ def send_key(self, keyname: str):
+
+        self._call_service(service="ui_service", action="launchapp", payload=launch)
+
+    def change_source(self, id: str):
+
+        if id == "0":
+            source = {"sourceid": "0"}
+
+        if id == "1":
+            source = {"sourceid": "1"}
+
+        if id == "2":
+            source = {"sourceid": "2"}
+
+        if id == "3":
+            source = {"sourceid": "3"}
+
+        if id == "4":
+            source = {"sourceid": "4"}
+
+        if id == "5":
+            source = {"sourceid": "5"}
+
+        if id == "6":
+            source = {"sourceid": "6"}
+
+        if id == "7":
+            source = {"sourceid": "7"}
```
This can be shortened:
```suggestion
    def change_source(self, id: str):
        payload = {"sourceid": id}
        self._call_service(service="ui_service", action="changesource", payload=payload)
```

newAM commented on hisensetv/__init__.py:
```diff
@@ -303,9 +370,64 @@ def send_key_play(self):
     @_check_connected
     def send_key_pause(self):
-        """ Sends a keypress of the stop key to the TV. """
+        """ Sends a keypress of the pause key to the TV. """
```
:+1:

newAM commented on hisensetv/__init__.py:
```diff
@@ -226,6 +239,60 @@ def send_key(self, keyname: str):
+
+        self._call_service(service="ui_service", action="launchapp", payload=launch)
+
+    def change_source(self, id: str):
```
Can you add a docstring here, or prefix with `_` to indicate that this is an internal function?

newAM commented on hisensetv/__main__.py:
```diff
@@ -4,12 +4,18 @@
 import ssl
 from . import HisenseTv
 
+choices_volume = []
+for testvar in range(0, 101):
+    choices_volume.append(str(testvar))
+
```
This appears to be unused?
```suggestion
```

newAM reviewed (changes requested):
Looks like the changes are passing CI now, left some comments for improvements!

JulianStremel committed:
Update hisensetv/__main__.py

Co-authored-by: Alex <7845120+newAM@users.noreply.github.com>

JulianStremel committed:
Update hisensetv/__init__.py

Co-authored-by: Alex <7845120+newAM@users.noreply.github.com>

JulianStremel committed:
Update hisensetv/__init__.py

Co-authored-by: Alex <7845120+newAM@users.noreply.github.com>

JulianStremel committed:
Better readability and extended set of keys

List of available keys:

power,
up,down,left,right,
menu,back,exit,ok,
volume_up,volume_down,
channel_up,channel_down,
fast_forward,rewind,
stop,play,pause,mute,
home,subtitle,
netflix,youtube,amazon,
0,1,2,3,4,5,6,7,8,9,
source_1,source_2,source_3,source_4,source_5,source_6,source_7",

JulianStremel committed:
Add files via upload

JulianStremel commented:
i added more keys and changed everything you highlighted , how can i commit these changes ?

newAM commented:
> i added more keys and changed everything you highlighted , how can i commit these changes ?

Should just be a standard `git commit` and `git push` (your repo, your branch).

Github has some tools to commit with the web interface, but I have never used that :fearful:
